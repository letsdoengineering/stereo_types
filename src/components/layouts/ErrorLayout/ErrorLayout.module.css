.vertical-center {
    position: absolute;
    width: 100%;
    top: 50%;
    transform: translateY(-50%);
}
