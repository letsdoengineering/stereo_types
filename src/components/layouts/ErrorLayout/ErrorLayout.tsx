import React from 'react'
import GridRow from '../../basics/Grid/GridRow/GridRow'
import GridColumn from '../../basics/Grid/GridColumn/GridColumn'
import Heading from '../../basics/Heading/Heading'
import Text from '../../basics/Text/Text'
import Spacing from '../../basics/Spacing/Spacing'
import List from '../../basics/List/List'
import ListItem from '../../basics/List/ListItem'
import Button from '../../basics/Button/Button'
import Link from '../../basics/Link/Link'
import * as errorLayoutStyles from './ErrorLayout.module.css'

type ErrorLayoutProps = {
    onClickGoBack(): void
}

const ErrorLayout: React.FC<ErrorLayoutProps> = ({ onClickGoBack }: ErrorLayoutProps) => {
    return (
        <>
            <GridRow className={errorLayoutStyles.verticalCenter} isPadded>
                <GridColumn />
                <GridColumn size='7'>
                    <Heading level='1' weight='Bold'>
                        Oops!
                    </Heading>
                    <Spacing size='Double' />
                    <Heading level='2' weight='Bold'>
                        Sorry, there seems to be a problem :(
                    </Heading>
                    <Spacing size='Double' />
                    <Text>
                        Some message to the user that says something or other and helps them along
                        their merry way back to a safe part of the site
                    </Text>
                    <Spacing size='Single' />
                    <Text>Here are some helpful links:</Text>
                    <Spacing size='Single' />
                    <List isIndented isLinks isUnstyled={false}>
                        <ListItem listKey='error-page-link-1'>
                            <Link href='' weight='Bold'>
                                Link 1
                            </Link>
                        </ListItem>
                        <ListItem listKey='error-page-link-2'>
                            <Link href='' weight='Bold'>
                                Link 2
                            </Link>
                        </ListItem>
                        <ListItem listKey='error-page-link-3'>
                            <Link href='' weight='Bold'>
                                Link 3
                            </Link>
                        </ListItem>
                    </List>
                    <Spacing size='Double' />
                    <GridRow>
                        <GridColumn size='6'>
                            <Button
                                onClick={onClickGoBack}
                                size='M'
                                buttonText='Go Back'
                                type='button'
                                fillSpace={false}
                            />
                        </GridColumn>
                        <GridColumn size='6' />
                    </GridRow>
                </GridColumn>
                <GridColumn />
            </GridRow>
        </>
    )
}

export default ErrorLayout
