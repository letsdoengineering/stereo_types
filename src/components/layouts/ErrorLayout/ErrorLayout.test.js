import * as React from 'react'
import ErrorLayout from './ErrorLayout'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import { axe } from 'jest-axe'

const mockContent = {
  footer: 'some footer',
}

describe('ErrorLayout', () => {
  it('has not changed the JSX layout of the page from previous snapshot', () => {
    const tree = renderer.create(<ErrorLayout content={mockContent} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders the page DOM that is accessible', async () => {
    const { container } = render(<ErrorLayout content={mockContent} />)
    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
