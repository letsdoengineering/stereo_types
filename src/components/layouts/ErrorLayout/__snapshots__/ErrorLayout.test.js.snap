// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorLayout has not changed the JSX layout of the page from previous snapshot 1`] = `
<div
  className="gridRow verticalCenter gridRowPadded"
>
  <div
    className="gridColumn sizeAuto"
  />
  <div
    className="gridColumn size7"
  >
    <h1
      className="title title1 weightBold"
    >
      Oops!
    </h1>
    <div
      className="spacingDouble spacing"
    />
    <h2
      className="title title2 weightBold"
    >
      Sorry, there seems to be a problem :(
    </h2>
    <div
      className="spacingDouble spacing"
    />
    <span
      className="sizeS weightNormal textGray"
    >
      Some message to the user that says something or other and helps them along their merry way back to a safe part of the site
    </span>
    <div
      className="spacingSingle spacing"
    />
    <span
      className="sizeS weightNormal textGray"
    >
      Here are some helpful links:
    </span>
    <div
      className="spacingSingle spacing"
    />
    <ul
      className="itemSpacing"
    >
      <li
        className="isLinks"
      >
        <a
          className="link sizeS weightBold"
          href=""
        >
          Link 1
        </a>
      </li>
      <li
        className="isLinks"
      >
        <a
          className="link sizeS weightBold"
          href=""
        >
          Link 2
        </a>
      </li>
      <li
        className="isLinks"
      >
        <a
          className="link sizeS weightBold"
          href=""
        >
          Link 3
        </a>
      </li>
    </ul>
    <div
      className="spacingDouble spacing"
    />
    <div
      className="gridRow"
    >
      <div
        className="gridColumn size6"
      >
        <button
          className="button sizeM"
          type="button"
        >
          Go Back
        </button>
      </div>
      <div
        className="gridColumn size6"
      />
    </div>
  </div>
  <div
    className="gridColumn sizeAuto"
  />
</div>
`;
